cpp/src/hash/md5_hash.cu

```diff
@@ -188,31 +188,29 @@ struct MD5Hash {
     // The message length is appended to the end of the last chunk processed
     uint64_t const message_length_in_bits = hash_state->message_length * 8;
 
-    auto const padding_begin = thrust::fill_n(thrust::seq,
-                                              hash_state->buffer + hash_state->buffer_length,
-                                              sizeof(end_of_message),
-                                              end_of_message);
-    auto const buffer_end    = hash_state->buffer + md5_chunk_size;
-    auto const message_end   = buffer_end - sizeof(message_length_in_bits);
-
-    if (padding_begin <= message_end) {
-      // The message size fits in this hash step. Pad up to the point where the message size
-      // goes with zeros.
-      thrust::fill(thrust::seq, padding_begin, message_end, 0x00);
-    } else {
+    auto padding_begin     = thrust::fill_n(thrust::seq,
+                                        hash_state->buffer + hash_state->buffer_length,
+                                        sizeof(end_of_message),
+                                        end_of_message);
```

Comment:
Can we just do this? Using a `fill` is a bit verbose for writing a single byte. 
```suggestion
   hash_state->buffer[hash_state->buffer_length - 1] = end_of_message;
```

Reply:
The alternative is slightly longer because we don't get the iterator as a result of `thrust::fill_n`.
```suggestion
    hash_state->buffer[hash_state->buffer_length] = end_of_message;
    auto padding_begin    = hash_state->buffer + hash_state->buffer_length + sizeof(end_of_message);
```

Reply:
`end_of_message` should probably be renamed, because it sounds like it could be an iterator instead of a sentinel value. 

How about this.

```suggestion
   auto message_end = hash_state->buffer + hash_state->buffer_length;
   *message_end = end_of_message;
```